packages/app/components/Today/TACheckinButton.tsx

```diff
@@ -0,0 +1,94 @@
+import { API } from "@template/api-client";
+import { Button, Input, Modal, Radio } from "antd";
+import { useRouter } from "next/router";
+import { ReactElement, useState } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+const CheckinButton = styled(Button)`
+  background: #2a9187;
+  border-radius: 6px;
+  color: white;
+  font-weight: 500;
+  font-size: 14px;
+`;
+
+export default function TACheckinButton({
+  courseId,
+}: {
+  courseId: number;
+}): ReactElement {
+  const router = useRouter();
+
+  const [viewCheckinModal, setViewCheckinModal] = useState(false);
+  const [value, setValue] = useState(0);
+  const [customRoom, setCustomRoom] = useState("");
+
+  const canSubmitCustomRoom = !(value === -1 && !customRoom);
+
+  const { data } = useSWR(
+    `api/v1/courses/${courseId}`,
+    async () => courseId && API.course.get(Number(courseId))
```

Comment:
Minor thing but if the props types expect this to be a number do we need to cast it in `API.course.get(Number(courseId))`?

Reply:
:).